### PR TITLE
Add jsonserver module

### PR DESCRIPTION
**Pull request by patrickelectric**
Signed-off-by: Patrick José Pereira <patrickelectric@gmail.com>

### CODE REVIEW

**Comment by patrickelectric** [comment]
This module broadcasts all status messages periodically via udp.

File: MAVProxy/modules/mavproxy_jsonserver.py [hunk 1]
```diff
@@ -0,0 +1,51 @@
...
+        self.sock = socket.socket(socket.AF_INET, socket.SOCK_DGRAM)
+
+        # Configure thread
+        server_thread = threading.Thread(target = self.loop)
+        server_thread.start()
+        server_thread.join()
+
+    def loop(self):
+        while True:
+            time.sleep(1)
```
Comment (jaxxzer) [inline MAVProxy/modules/mavproxy_jsonserver.py]: why 1Hz limiting? I'd like 1kHz.

**Comment by jaxxzer** [comment]
When you say status, I see this refers to `mpstate.status`. Does this hold all incoming msgs? Or only SYS_STATUS or STATUSTEXT or what?

**Comment by patrickelectric** [comment]
@jaxxzer, Added address and freq parameters.

It'll send `mpstate.status` content.
You can check the json [here](https://jsonblob.com/93280d93-4d9d-11e8-ad5f-0767cabf2e75). It was created from ardusub sitl.

**Comment by patrickelectric** [comment]
Here is a simple python script to test it:
```python
import json
import select
import socket

UDP_IP = "127.0.0.1"
UDP_PORT = 4777

sock = socket.socket(socket.AF_INET, socket.SOCK_DGRAM)
sock.bind((UDP_IP, UDP_PORT))
sock.setblocking(0)

while True:
    r, _, _ = select.select([sock],[],[],0)
    if not r:
        pass
    if r:
        s = json.loads(sock.recv(10240).decode('utf-8'))
        print(s['LOCAL_POSITION_NED']['time_boot_ms'])
```

**Comment by patrickelectric** [comment]
@peterbarker I don't know, I put the output of this module in the previous comment. Do you have one of APWeb to compare ?

**Comment by patrickelectric** [comment]
Closing this PR, it'll be created another one with json and rest.

**Pull request closed without merging**